drivers/storage/portworx/components_test.go

```diff
@@ -5830,6 +5831,75 @@ func TestCSI_0_3_NodeAffinityChange(t *testing.T) {
 	require.Nil(t, csiStatefulSet.Spec.Template.Spec.Affinity)
 }
 
+func TestPrometheusUpgradeDefaultDesiredImages(t *testing.T) {
+	versionClient := fakek8sclient.NewSimpleClientset()
+	coreops.SetInstance(coreops.New(versionClient))
+	versionClient.Discovery().(*fakediscovery.FakeDiscovery).FakedServerVersion = &version.Info{
+		GitVersion: "v1.22.0",
+	}
+	reregisterComponents()
+	k8sClient := testutil.FakeK8sClient()
+	driver := portworx{}
+	driver.Init(k8sClient, runtime.NewScheme(), record.NewFakeRecorder(0))
+
+	cluster := &corev1.StorageCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "px-cluster",
+			Namespace: "kube-test",
+			Annotations: map[string]string{
+				pxutil.AnnotationPVCController: "true",
+			},
+		},
+		Spec: corev1.StorageClusterSpec{
+			Monitoring: &corev1.MonitoringSpec{
+				Prometheus: &corev1.PrometheusSpec{
+					Enabled: true,
+					AlertManager: &corev1.AlertManagerSpec{
+						Enabled: true,
+					},
+				},
+			},
+		},
+		Status: corev1.StorageClusterStatus{
+			DesiredImages: &corev1.ComponentImages{
+				PrometheusOperator: manifest.DefaultPrometheusOperatorImage,
+			},
+		},
+	}
+
+	// TestCase: k8s 1.22 should upgrade Prometheus operator from 0.34.0 to 0.50.0
+	m := &fakeManifest{}
+	m.Init(nil, nil, k8sutil.K8sVer1_22)
+	manifest.SetInstance(m)
+	defer manifestSetup()
+
+	driver.SetDefaultsOnStorageCluster(cluster)
+	require.Equal(t, "quay.io/prometheus-operator/prometheus-operator:v0.50.0", cluster.Status.DesiredImages.PrometheusOperator)
+
+	k8sClient.Create(
+		context.TODO(),
+		&v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      component.AlertManagerConfigSecretName,
+				Namespace: cluster.Namespace,
+			},
+		},
+		&client.CreateOptions{},
+	)
+
+	err := driver.PreInstall(cluster)
+	require.NoError(t, err)
+
+	// Prometheus operator ClusterRole should be updated as well
+	expectedCR := testutil.GetExpectedClusterRole(t, "prometheusNewOperatorClusterRole.yaml")
```

Comment:
Can we verify the prometheus-operator deployment too without the config map reload argument and the correct images?

Reply:
We should even check that the CRDs have been registered.

Reply:
Can we rename it to `prometheusNewOperatorClusterRole-v0.50.0.yaml`.
I feel we need to discuss a version management system in the future for the operator. I see many components' version scattering everywhere in the codes.